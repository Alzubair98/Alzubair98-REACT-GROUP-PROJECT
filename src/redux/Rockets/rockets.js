import axios from 'axios';
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const LODING = 'LODING';

const url = 'https://api.spacexdata.com/v3/rockets';

export const loadRockets = createAsyncThunk(LODING, async () => {
  const response = await axios.get(url);
  const res = response.data;
  const data = res.map((item) => ({
    id: item.id,
    name: item.rocket_name,
    description: item.description,
    img: item.flickr_images[0],
  }));
  return data;
});

const storeSlice = createSlice({
  name: 'space Hup',
  initialState: [],
  extraReducers: {
    [loadRockets.fulfilled]: (state, action) => action.payload,
  },
});

export default storeSlice.reducer;
