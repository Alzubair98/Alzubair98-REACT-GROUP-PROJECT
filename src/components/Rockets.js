import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import RocketCard from './RocketCard';
import { loadRockets } from '../redux/Rockets/rockets';

const Rockets = () => {
  const state = useSelector((state) => state.storeSlice);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadRockets());
  }, []);

  return (
    <>
      <div>
        {state.map((rocket) => (
          <RocketCard
            key={rocket.id}
            name={rocket.name}
            src={rocket.img}
            desc={rocket.description}
          />
        ))}

      </div>
    </>
  );
};

export default Rockets;
