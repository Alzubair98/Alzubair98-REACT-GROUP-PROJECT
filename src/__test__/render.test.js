import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/Rockets';
import Profile from '../components/profile';
import store from '../redux/configureStore';

describe('Rockets & Profile renderer test', () => {
  it('match snapshot for Rockets', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('match snapshot for the Profile', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
