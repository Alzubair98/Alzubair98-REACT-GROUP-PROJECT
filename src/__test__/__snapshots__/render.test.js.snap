// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets & Profile renderer test match snapshot for Rockets 1`] = `<div />`;

exports[`Rockets & Profile renderer test match snapshot for the Profile 1`] = `
<div
  className="profile-cont"
>
  <div>
    mission
  </div>
  <div
    className="rockets-profile"
  >
    <h1>
      My Rockets
    </h1>
  </div>
</div>
`;
